README.md

```diff
@@ -263,22 +262,48 @@ $ scl scenario build scenarios/loop --clean
 # add --headless if you do not need visualisation
 $ python examples/single_agent.py scenarios/loop
 
-# On your host machine visit http://localhost:8081 to see the running simulation in
-# Envision.
+# On your host machine visit http://localhost:8081 to see the running simulation in Envision.
 ```
 
 (For those who have permissions:) if you want to push new images to our [public dockerhub registry](https://hub.docker.com/orgs/huaweinoah) run,
 
 ```bash
 # For this to work, your account needs to be added to the huaweinoah org
-$ cd /path/to/SMARTS
+$ cd </path/to/SMARTS>
 export VERSION=v0.4.17
 docker build --no-cache -f ./utils/docker/Dockerfile -t smarts:$VERSION .
 docker tag smarts:$VERSION huaweinoah/smarts:$VERSION
 docker login
 docker push huaweinoah/smarts:$VERSION
 ```
 
+### Using Singularity
+```bash
+$ cd </path/to/SMARTS>
+
+# Build container from definition file.
+$ sudo singularity build ./utils/singularity/smarts.sif ./utils/singularity/smarts.def
+
+# Use the container to build the required scenarios.
+$ singularity shell --containall --bind ../SMARTS:/SMARTS ./utils/singularity/smarts.sif
+# Inside the container
+$ scl scenario build /SMARTS/scenarios/loop/
+$ exit
+
+# Then, run the container using one of the following methods.
+
+# 1. Run container in interactive mode.
+$ singularity shell --containall --bind ../SMARTS:/SMARTS ./utils/singularity/smarts.sif
+# Inside the container
+$ python3.7 /SMARTS/examples/single_agent.py /SMARTS/scenarios/loop/ --headless
```

Comment:
```suggestion
(singularity) $ python3.7 /SMARTS/examples/single_agent.py /SMARTS/scenarios/loop/ --headless
```

Reply:
I have changed `$` to be `Singularity>`, in order to look the same as it appears on the terminal.